src/app/pages/rpc-stx-transfer-sip10-ft/rpc-stx-transfer-sip10-ft.utils.ts

```diff
@@ -0,0 +1,82 @@
+import type { StacksNetwork } from '@stacks/network';
+import {
+  type ClarityValue,
+  createAddress,
+  noneCV,
+  postConditionToWire,
+  serializeCV,
+  serializePostConditionWire,
+  standardPrincipalCVFromAddress,
+  uintCV,
+} from '@stacks/transactions';
+
+import { createRequestEncoder, stxTransferSip10Ft } from '@leather.io/rpc';
+import {
+  type StacksUnsignedContractCallOptions,
+  TransactionTypes,
+  getStacksAssetStringParts,
+} from '@leather.io/stacks';
+import { createMoney } from '@leather.io/utils';
+
+import { getAddressFromAssetString } from '@shared/utils';
+import { makeFtPostCondition } from '@shared/utils/post-conditions';
+
+import { initialSearchParams } from '@app/common/initial-search-params';
+
+export function getDecodedRpcStxTransferSip10FtRequest() {
+  const { decode } = createRequestEncoder(stxTransferSip10Ft.request);
+  const rpcRequest = initialSearchParams.get('rpcRequest');
+  if (!rpcRequest) throw new Error('Missing rpcRequest');
+  return decode(rpcRequest);
+}
+
+export type RpcStxTransferSip10FtRequest = ReturnType<
+  typeof getDecodedRpcStxTransferSip10FtRequest
+>;
+
+interface GetStacksUnsignedContractCallOptionsFromRequestArgs {
+  address: string;
+  publicKey: string;
+  network: StacksNetwork;
+}
+export function getStacksUnsignedContractCallOptionsFromRequest({
+  address,
+  publicKey,
+  network,
+}: GetStacksUnsignedContractCallOptionsFromRequestArgs): StacksUnsignedContractCallOptions {
+  const decodedRpcRequest = getDecodedRpcStxTransferSip10FtRequest();
+
+  const { contractAddress, contractAssetName, contractName } = getStacksAssetStringParts(
+    decodedRpcRequest.params.asset
+  );
+
+  const fnArgs: ClarityValue[] = [
+    uintCV(decodedRpcRequest.params.amount),
+    standardPrincipalCVFromAddress(createAddress(decodedRpcRequest.params.address ?? address)),
+    standardPrincipalCVFromAddress(createAddress(decodedRpcRequest.params.recipient)),
+    noneCV(),
+  ];
+  const postConditionOptions = {
+    amount: decodedRpcRequest.params.amount,
+    contractAddress,
+    contractAssetName,
+    contractName,
+    stxAddress: decodedRpcRequest.params.address ?? address,
+  };
+
+  return {
+    txType: TransactionTypes.ContractCall,
+    contractAddress: getAddressFromAssetString(decodedRpcRequest.params.asset),
+    contractName,
+    functionArgs: fnArgs.map(arg => serializeCV(arg)),
+    functionName: 'transfer',
+    postConditions: [
+      serializePostConditionWire(postConditionToWire(makeFtPostCondition(postConditionOptions))),
+    ],
+    fee: createMoney(decodedRpcRequest.params.fee ?? 0, 'STX'),
+    nonce: decodedRpcRequest.params.nonce ?? 0,
```

Comment:
Is there any other way we can avoid doing these `??`s? In this fn, can't we leave the value undefined if not passed in? 

I know it can be hard to get around typing issues, but this just sets us up for bugs where nonce defauts to zero.

Reply:
If the type doesn't allow for `undefined` from `@stacks/transactions`, any suggestions? I'll try to research it today to see if I can find a better pattern to use.

Reply:
I believe for the fee estimation we need them defined, but will check. In our stacks pkg we actually create new types and require them for this reason, otherwise I was wrong bc the `@stacks/transactions` type has them as optional. I will double check this though, but I'm pretty sure we get errors if they aren't defined.

Reply:
They def need to be defined at the point when we create the tx, but don't we have a loader component for that?

Reply:
We have to gen an initial unsignedTx in the fees loader in order to get the Stacks fee estimates. The nonce should be fine using the loader, but we will get a white screen like this if we don't have a default fee. We could just use a default fee in the fees loader though to gen the initial fees for the editor? The api doesn't really need an accurate fee to give us the fee estimations, right?

![Screenshot 2025-05-06 at 12 48 26 PM](https://github.com/user-attachments/assets/03d2f3b0-e4a9-4182-9a39-51576871ea9f)

Reply:
If I make the types optional and just do this in the mono repo, I can get past the type errors for undefined and the white screen for fee estimates...

<img width="519" alt="Screenshot 2025-05-06 at 12 58 26 PM" src="https://github.com/user-attachments/assets/646805c3-aa0a-4b3b-96c8-b87033989bfd" />

Reply:
This approach also works in the fees loader and then nothing has to be set as a default generating the unsignedTx overall...

<img width="781" alt="Screenshot 2025-05-06 at 1 03 07 PM" src="https://github.com/user-attachments/assets/5c898001-c8f3-4e5e-9d8a-067b5ca9716a" />

Reply:
> We have to gen an initial unsignedTx in the fees loader in order to get the Stacks fee estimates.

Could this be separated out, such that where we pass the `?? 0` is very clearly only for the tx fee estimation? Such that, where we call the same fn for the actual tx, we don't have any zero fall backs?

Reply:
I need these changes installed here: https://github.com/leather-io/mono/pull/1148

Reply:
Commented on other PR, sorry for back forward here, I'm just wondering if there's a way to handle the fallback values in a very contained manner that doesn't impact other tx construction logic